src/Microsoft.OData.Core/UrlValidation/ODataUrlValidationError.cs

```diff
@@ -0,0 +1,71 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUrlValidationError.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.OData.UriParser.Validation
+{
+    /// <summary>
+    /// Severity of an OData Url Validation Error
+    /// </summary>
+    public enum Severity
+    {
+        Warning,
+        Error
+    }
+
+    /// <summary>
+    /// A class representing an error encountered while validating an OData Url.
+    /// </summary>
+    public class ODataUrlValidationError
+    {
+        private Dictionary<string, object> additionalInfo;
+
+        /// <summary>
+        /// The error code.
+        /// </summary>
+        public string ErrorCode;

```

Comment:
Would suggest to make these read-only auto properties. 
```suggestion
        public string ErrorCode { get; }
```

Reply:
makes sense; can't think of a reason for changing the values once the error is created.

---
In reply to: [424640047](https://github.com/OData/odata.net/pull/1769#discussion_r424640047) [](ancestors = 424640047)